Survey/Survey/Sources/Presentation/Modules/Login/LoginViewModel.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  LoginViewModel.swift
+//  Survey
+//
+//  Created by Taher on 7/11/22.
+//
+
+import Combine
+import SwiftUI
+
+final class LoginViewModel: ObservableObject {
+
+    // MARK: Input
+
+    @Published var email: String = .empty
+    @Published var password: String = .empty
+    var didTapLoginButton = PassthroughSubject<Void, Never>()
+
+    // MARK: Output
+
+    @Published var errorMessage: String = .empty
+
+    private let loginUseCase: LoginUseCaseProtocol
+    private var disposables = Set<AnyCancellable>()
+
+    init(loginUseCase: LoginUseCase) {
+        self.loginUseCase = loginUseCase
+
+        didTapLoginButton.sink { [weak self] _ in
+            // TODO: check email pattern validity in integration task
+            self?.login()
+        }
+        .store(in: &disposables)
+    }
+
+    private func login() {
```

Comment:
IMO, since we won't have separate `input` and `output` like we usually do when implementing MVVM with RxSwift, I think `didTapLoginButton` is not necessary. Instead we can public `login()`, then on the `LoginView` just:
```
Button(action: { viewModel.login() } ...
```

Reply:
updated in this [commit](https://github.com/Shayokh144/ic-survey-swiftui-combine/pull/43/commits/1aab549f658f6030d71ed30af0d2429788c73d4b)